test/org/opensolaris/opengrok/web/DirectoryListingTest.java

```diff
@@ -238,25 +257,34 @@ private long getLastModified(Node item) throws Exception {
 
         String value = val.getNodeValue();
         return value.equalsIgnoreCase("Today")
-            ? Long.MAX_VALUE
-            : dateFormatter.parse(value).getTime();
+                ? Long.MAX_VALUE
+                : dateFormatter.parse(value).getTime();
     }
 
     /**
      * Get the size from the: &lt;td&gt;&lt;tt&gt;size&lt;/tt&gt;&lt;/td&gt;
+     *
      * @param item the node representing &lt;td&gt;
      * @return The size
-     * @throws java.lang.Exception if an error occurs
      */
-    private int getSize(Node item) throws Exception {
+    private int getSize(Node item) throws NumberFormatException {
         Node val = item.getFirstChild();
         assertNotNull(val);
         assertEquals(Node.TEXT_NODE, val.getNodeType());
-        return Integer.parseInt(val.getNodeValue().trim());
+        try {
+            return Integer.parseInt(val.getNodeValue().trim());
+        } catch (NumberFormatException ex) {
+            if (DirectoryListing.DIRECTORY_SIZE_PLACEHOLDER.equals(val.getNodeValue().trim())) {
```

Comment:
why not check the `-` equality first and then try to convert to integer ? Checking the character only after exception looks a bit convoluted.